### PR TITLE
Users

### PR DESCRIPTION
**Pull request by urumo**
## Summary by CodeRabbit

- **New Features**
	- Introduced user management functionalities through a new `UsersController`, allowing user creation, authentication, and retrieval.
	- Added JWT authentication configuration to enhance security.
	- Implemented a new `UserManagerService` for JWT generation and user validation.
	- Added a new `Hello` grain for managing greetings with persistence.
	- Created an interface `IHello` for structured grain functionalities.
	- Enhanced application configuration with new JWT settings.

- **Bug Fixes**
	- Corrected formatting and structure in various configuration files for improved readability and maintainability.

- **Refactor**
	- Removed obsolete projects and configurations, streamlining the application structure.
	- Updated service registrations and middleware configurations to reflect architectural changes.

- **Documentation**
	- Enhanced the configuration settings for JWT in the `appsettings.Development.json` file.

### CODE REVIEW

**Comment by urumo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by urumo** [comment]
@coderabbitai review

File: src/AppHost/Program.cs [hunk 1]
```diff
@@ -24,6 +25,10 @@
     .WithEnvironment("sfu__url", sfuUrl)
     .WithEnvironment("sfu__clientId", sfuClientId)
     .WithEnvironment("sfu__clientSecret", sfuClientSecret)
+    .WithEnvironment("Jwt__Issuer", "Arhon")
+    .WithEnvironment("Jwt__Audience", "Arhon")
```
Comment (0xF6) [inline src/AppHost/Program.cs:29]: Арчон?

File: src/Argon.Api/Extensions/JwtExtension.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }).AddJwtBearer(o =>
+        {
+            o.TokenValidationParameters = new TokenValidationParameters
+            {
+                ValidIssuer = builder.Configuration["Jwt:Issuer"],
+                ValidAudience = builder.Configuration["Jwt:Audience"],
+                IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Jwt:Key"])),
+                ValidateIssuer = true,
+                ValidateAudience = true,
+                ValidateLifetime = false,
```
Comment (0xF6) [inline src/Argon.Api/Extensions/JwtExtension.cs:25]:
```suggestion
                ValidateLifetime = true,
```

File: src/Argon.Api/Grains/UserManager.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public async Task<UserStorageDto> Create(string password)
+    {
+        var username = this.GetPrimaryKeyString();
+        await EnsureUnique();
+        await managerService.Validate(username, password);
+
+        userStore.State.Id = Guid.NewGuid();
+        userStore.State.Username = username;
+        userStore.State.Password = BCrypt.HashPassword(password);
```
Comment (0xF6) [inline src/Argon.Api/Grains/UserManager.cs:23]: пароль без соли?

**Review by 0xF6 — CHANGES_REQUESTED** [review]

File: src/AppHost/Program.cs (hunk 1 above)
Comment (urumo) [reply src/AppHost/Program.cs:29]: Арон

File: src/Argon.Api/Extensions/JwtExtension.cs (hunk 2 above)
Comment (urumo) [reply src/Argon.Api/Extensions/JwtExtension.cs:25]: тупанул внатуре

File: src/Argon.Api/Grains/UserManager.cs (hunk 3 above)
Comment (urumo) [reply src/Argon.Api/Grains/UserManager.cs:23]: я не в питере

**Pull request merged**